extensions/wikia/SpitfiresContributionExperiments/Hooks.class.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Wikia\ContributionExperiments;
+
+class Hooks {
+
+	const NEWLY_REGISTERED_USER = 'newlyregistered';
+	const WITHOUT_EDIT_USER = 'userwithoutedit';
+
+	const COOKIE_EXPERIMENT_TIME = 60 * 60 * 24 * 30;
+
+	/**
+	 * Register hooks for the extension
+	 */
+	public static function register() {
+		$hooks = new self();
+		\Hooks::register( 'UserSignupAfterSignupBeforeRedirect', [ $hooks, 'onUserSignupAfterSignupBeforeRedirect' ] );
+		\Hooks::register( 'UserLoginComplete', [ $hooks, 'onUserLoginComplete' ] );
+		\Hooks::register( 'UserLoadFromHeliosToken', [ $hooks, 'onUserLoadFromHeliosToken' ] );
+		\Hooks::register( 'BeforePageDisplay', [ $hooks, 'onBeforePageDisplay' ] );
+	}
+
+	public function onUserSignupAfterSignupBeforeRedirect( $redirectUrl ) {
+		$this->setCookie( self::NEWLY_REGISTERED_USER, 1, time() + self::COOKIE_EXPERIMENT_TIME );
+
+		return true;
+	}
+
+	public function onUserLoginComplete( \User $user, $html ) {
+		$this->manageUserActivityGroupCookie( $user );
+
+		return true;
+	}
+
+	public function onUserLoadFromHeliosToken( \User $user ) {
+		$this->manageUserActivityGroupCookie( $user );
+
+		return true;
+	}
+
+	public function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
+		$user = \RequestContext::getMain()->getUser();
```

Comment:
`OutputPage` is a context source, so you can just use `$out->getUser();` (and actually `Skin` is too so `$skin->getUser();` would also work :)).